java/vector/src/main/java/org/apache/arrow/vector/ValueVector.java

```diff
@@ -106,6 +107,8 @@
 
   TransferPair getTransferPair(String ref, BufferAllocator allocator);
 
+  TransferPair getTransferPair(String ref, BufferAllocator allocator, CallBack callBack);
```

Comment:
why don't we just add a parameter to the existing method?
We should avoid duplicating those methods unless necessary.
Please add javadoc to interface/abstract methods